sklearn/manifold/spectral_embedding_.py

```diff
@@ -44,14 +44,18 @@ def _graph_connected_component(graph, node_id):
     """
     connected_components_matrix = np.zeros(
         shape=(graph.shape[0]), dtype=np.bool)
-    connected_components_matrix[node_id] = True
+    nodes_to_explore = np.zeros(shape=(graph.shape[0]), dtype=np.bool)
+    nodes_to_explore[node_id] = True
     n_node = graph.shape[0]
     for i in range(n_node):
-        last_num_component = connected_components_matrix.sum()
-        _, node_to_add = np.where(graph[connected_components_matrix] != 0)
-        connected_components_matrix[node_to_add] = True
-        if last_num_component >= connected_components_matrix.sum():
+        nodes_to_add = np.zeros(shape=(graph.shape[0]), dtype=np.bool)
```

Comment:
I am wondering if the following would not be slightly more efficient (probably depends of the malloc used by numpy):

```python
nodes_to_add = np.empty(shape=(graph.shape[0]), dtype=np.bool)
for i in range(n_node):
    nodes_to_add.fill(False)
    ...
```